### PR TITLE
feature: integrate feature-unavailable modal for keycard

### LINKED ISSUE #21755: Sync is not available for mobile keycard accounts
# Bug Report

## Problem 
As there is no option to sync current mobile keycard wallet with desktop, users cannot continue the flows that are not available on mobile

#### Expected behavior
can sync (or use a fallback feature), see all accounts on desktop and continue to use it.

#### Actual behavior
cannot sync, as Showing QR code is blocked
Image

### Reproduction

1. Open profile > sync
2. Try to generate new QR

### Additional Information

- Status version: nightly 
- Operating System: Android, iOS

[comment]: # (Please, add logs/notes if necessary)

### LINKED ISSUE #21778: Keycard - Disable or hide unsupported Keycard flows
# Problem

In Status 2.32 Keycard users won't be able to use their keycards in all possible flows. We want to provide a better user experience to those users so that they don't waste time in flows that will surely fail at some point.

## Acceptance Criteria

In case a Profile is set-up with a Keycard: 

- Request to join a community should show a drawer stopping the user from proceeding https://www.figma.com/design/YGm3igIOAcwMqUVJWCJ6f1/Keycard?node-id=4588-67142&node-type=instance&m=dev
- Show the special _unsupported feature_ drawer when the user tries to **enable** biometrics.
- Hide option to edit password.

## Out of scope

- Ignore editing shared addresses in Figma https://www.figma.com/design/YGm3igIOAcwMqUVJWCJ6f1/Keycard?node-id=4588-104535&node-type=shape_with_text&m=dev (this feature is disabled behind a feature flag, so we don't need to worry about it).
- Wallet-related flows.

## Resources

- Check PR https://github.com/status-im/status-mobile/pull/21771 because this issue can reuse some of the code related to the drawer.

### PR DESCRIPTION
**Pull request by seanstrom**
fixes https://github.com/status-im/status-mobile/issues/21778
fixes https://github.com/status-im/status-mobile/issues/21755

### Summary

* This PR hides the change password menu and button from keycard users.
* This PR attempts to implement a modal for show displaying a feature is unavailable, and this PR attempts to integrate with this modal with specific features like:
  * Request to Join Community
  * Enabling Biometrics
  * Adding a Paired Device
  * And any other place that uses standard-authentication with keycard-support disabled

#### Platforms

- Android
- iOS

##### Functional

- Settings - Sync
- Settings - Password
- Community - Overview

### Steps to test

- Open Status mobile app
- Sign Up or Login with Keycard Profile
- Open the Settings screen and view the Password Settings
  - Attempt to enable the biometrics
- Open the Settings screen and view the Sync settings, Paired devices
  - Attempt to add a paired device
- Open the Discover Communities screen, open Status community
  - Attempt to join the community

### Before and after screenshots comparison

| Description | Screen |
| --- | --- |
| Change Password is hidden from keycard users | IMG_0021 |
| Enabling Biometrics is unavailable to keycard users | IMG_0022 |
| Adding Paired Device is unavailable to keycard users | IMG_0018 |
| Revealing Sync Code is unavailable to keycard users. This is an example of how the standard authentication will now show this modal when keycard-support is not enabled. | IMG_0019 |

status: ready

### CODE REVIEW

File: src/status_im/contexts/keycard/feature_unavailable/view.cljs [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+   [quo/information-box
+    {:type  :default
+     :icon  :i/info
+     :style {:margin-top 8 :margin-horizontal 20}}
+    [:<>
+     (i18n/label :t/feature-unavailable-info)
+     [quo/text
+      {:style    {:text-decoration-line :underline}
+       :size     :paragraph-2
+
```
Comment (ilmotta) [inline src/status_im/contexts/keycard/feature_unavailable/view.cljs]: Extra lines

File: src/status_im/contexts/syncing/syncing_devices_list/view.cljs [hunk 2]
```diff
@@ -26,19 +31,24 @@
...
+  (let [theme                                       (quo.theme/use-theme)
+        devices                                     (rf/sub [:pairing/installations])
         devices-with-button                         (map #(assoc % :show-button? true) devices)
         user-device                                 (first devices-with-button)
         other-devices                               (rest devices-with-button)
         profile-color                               (rf/sub [:profile/customization-color])
         open-setup-syncing-with-customization-color (rn/use-callback (partial open-setup-syncing
                                                                               profile-color)
                                                                      [profile-color])
+        show-feature-unavailable-with-theme         (rn/use-callback (partial show-feature-unavailable
```
Comment (ilmotta) [inline src/status_im/contexts/syncing/syncing_devices_list/view.cljs]: A little nitpick @seanstrom:

I found this usage of `partial` and the other above it (not from this PR) confusing because `show-feature-unavailable` only expects one argument `theme` (but I had to check the function definition to be sure), thus there are no more args remaining from the partial application. Here `partial` is used only to generate an anonymous function, but this is quite uncommon in Clojure. If the implementation was `(rn/use-callback #(show-feature-available theme))` I would immediately know there was nothing else to understand about this callback.

**Review by ilmotta — APPROVED** [review]

**Comment by ilmotta** [comment]
@seanstrom Thank you for the detailed PR description. This is incredibly helpful for reviewers and QAs and even for the dev to make sure all requirements are covered 💯

**Comment by ilmotta** [comment]
@xAlisher @seanstrom, after reviewing this PR I started to wonder if the `Settings > Password` section shouldn't be renamed in 2.33. For a profile set-up with a keycard, the word _password_ loses its meaning a bit. The section could be renamed to `Authentication` or even just `Auth`, to give one alternative.

**Comment by flexsurfer** [comment]
hey @seanstrom have you seen this one https://github.com/status-im/status-mobile/pull/21771 , cc 
@mohsen-ghafouri

File: src/status_im/common/standard_authentication/events.cljs [hunk 3]
```diff
@@ -15,19 +15,14 @@
    (when on-auth-success (on-auth-success ""))))
 
 (defn authorize
-  [{:keys [db]} [{:keys [on-auth-success keycard-supported?] :as args}]]
+  [{:keys [db]} [{:keys [on-auth-success keycard-supported? theme] :as args}]]
```
Comment (flexsurfer) [inline src/status_im/common/standard_authentication/events.cljs:18]: could you pls revert this change because this popup was removed already in my PR, thanks

**Comment by flexsurfer** [comment]
hey @seanstrom i've merged @mohsen-ghafouri 's PR, and rebased my PR on it, and moved your changes there https://github.com/status-im/status-mobile/pull/21753/commits/37c73d0553c79f56a5c729b844f7359a2fb7b097, so we can close this PR ,thanks
